component/discovery/triton/triton.go

```diff
@@ -0,0 +1,79 @@
+package triton
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/prometheus/common/model"
+	prom_discovery "github.com/prometheus/prometheus/discovery/triton"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.triton",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Arguments struct {
+	Account         string           `river:"account,attr"`
+	Role            string           `river:"role,attr,optional"`
+	DNSSuffix       string           `river:"dns_suffix,attr"`
+	Endpoint        string           `river:"endpoint,attr"`
+	Groups          []string         `river:"groups,attr,optional"`
+	Port            int              `river:"port,attr,optional"`
+	RefreshInterval time.Duration    `river:"refresh_interval,attr,optional"`
+	Version         int              `river:"version,attr,optional"`
+	TLSConfig       config.TLSConfig `river:"tls_config,block,optional"`
```

Comment:
Most of the discovery has the full HttpClientConfig. Is there a reason this only has tls?

Reply:
ya, that's what prometheus supports for this discovery. if I should still be using HttpClientConfig let me know but I'm not sure how it would work in this case.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/#triton_sd_config

https://github.com/prometheus/prometheus/blob/c5e1498ae243bbf0b8363786de065bf98acd1edb/discovery/triton/triton.go#L68